Examples/GenerativeAICLI/Sources/GenerateContent.swift

```diff
@@ -70,7 +76,32 @@ struct GenerateContent: AsyncParsableCommand {
         name: modelNameOrDefault(),
         apiKey: apiKey,
         generationConfig: config,
-        safetySettings: safetySettings
+        safetySettings: safetySettings,
+        tools: [Tool(functionDeclarations: [
+          FunctionDeclaration(
+            name: "get_exchange_rate",
+            description: "Get the exchange rate for currencies between countries",
+            parameters: Schema(
+              type: .object,
+              properties: [
+                "currency_from": Schema(
+                  type: .string,
+                  format: "enum",
+                  description: "The currency to convert from in ISO 4217 format",
+                  enumValues: ["USD", "EUR", "JPY", "GBP", "AUD", "CAD"]
+                ),
+                "currency_to": Schema(
+                  type: .string,
+                  format: "enum",
+                  description: "The currency to convert to in ISO 4217 format",
+                  enumValues: ["USD", "EUR", "JPY", "GBP", "AUD", "CAD"]
+                ),
+              ],
+              required: ["currency_from", "currency_to"]
+            )
+          ),
+        ])],
+        requestOptions: RequestOptions(apiVersion: "v1beta")
```

Comment:
I think is a good approach that may allow for further abstraction. I'm not fully convinced of this, but perhaps introducing a `ParameterDeclaration` type may avoid potential confusion/complexity introduced by the nested `Schema` type.

Reply:
I think it might also be worth having a type for the top-level schema, which must be of type `.object`. A developer might be tempted to specify, e.g., `Schema(type: .number, format: "int64", description: "My Int parameter")` for a function that only takes one parameter but it will be rejected by the backend.

Reply:
I added a possible approach for the top-level schema in #117.

I'm not quite sure how to approach the nested schemas since quite elaborate forms with nested objects and arrays are supported (each with their own schemas), e.g.:
```
Schema(
  type: .object,
  properties: ["conversion_details": Schema(
    type: .object,
    description: "Details of the currency conversion to perform.",
    properties: [
      "from_currency": Schema(
        type: .string,
        format: "enum",
        description: "The currency to convert from in ISO 4217 format",
        enumValues: ["USD", "EUR", "JPY", "GBP", "AUD", "CAD"]
      ),
      "to_currencies": Schema(
        type: .array,
        items: Schema(
          type: .string,
          format: "enum",
          description: "The currencies to convert to in ISO 4217 format",
          enumValues: ["USD", "EUR", "JPY", "GBP", "AUD", "CAD"]
        )),
    ],
    requiredProperties: ["from_currency", "to_currency"]
  )],
  requiredProperties: ["conversion_details"]
)
```

Which would result in a function call payload from the backend like:
```
{
  "conversion_details": {
    "from_currency": "EUR",
    "to_currencies": ["USD", "AUD", "CAD"]
  }
}
```